src/py/mat3ra/made/tools/build/interface/builders.py

```diff
@@ -189,3 +197,192 @@ def _post_process(self, items: List[_GeneratedItemType], post_process_parameters
         for material, strain in zip(materials, strains):
             material.metadata["mean_abs_strain"] = strain
         return materials
+
+
+########################################################################################
+#                       Twisted Interface Builders                                     #
+########################################################################################
+
+
+class TwistedInterfaceConfiguration(BaseConfiguration):
+    film: Material
+    substrate: Material
+    twist_angle: float = Field(0, description="Twist angle in degrees")
+    distance_z: float = 3.0
+
+    @property
+    def _json(self):
+        return {
+            "type": self.get_cls_name(),
+            "film": self.film.to_json(),
+            "substrate": self.substrate.to_json(),
+            "twist_angle": self.twist_angle,
+            "distance_z": self.distance_z,
+        }
+
+
+class NanoRibbonTwistedInterfaceConfiguration(TwistedInterfaceConfiguration):
+    """
+    Configuration for creating a twisted interface between two materials using nanoribbons.
+
+    Args:
+        film (Material): The film material.
+        substrate (Material): The substrate material.
+        twist_angle (float): Twist angle in degrees.
+        ribbon_width (int): Width of the nanoribbon in unit cells.
+        ribbon_length (int): Length of the nanoribbon in unit cells.
+        distance_z (float): Vertical distance between layers in Angstroms.
+        vacuum_x (int): Vacuum padding in x direction in unit cells.
+        vacuum_y (int): Vacuum padding in y direction in unit cells.
```

Comment:
Vacuum padding should be calculated based on the shape of the resulting material